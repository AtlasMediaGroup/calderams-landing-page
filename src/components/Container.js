import React from 'react'
import PropTypes from 'prop-types'
import theme from '../theme'
import { compose } from 'glamor'

const styles = {
  container: {
    boxSizing: 'border-box',
    marginLeft: 'auto',
    marginRight: 'auto',
    paddingLeft: '2rem',
    paddingRight: '2rem',
    position: 'relative',

    [theme.breakpoint.mediumUp]: {
      paddingLeft: '2rem',
      paddingRight: '2rem'
    }
  }
}

export default function Container({ width, ...props }) {
  const size = !isNaN ? width : theme.container[width]

  //   return <div css={[styles.container, { maxWidth: size }]} {...props} />
  return (
    <div
      className={compose(
        styles.container,
        { maxWidth: size }
      )}
      {...props}
    />
  )
}

Container.propTypes = {
  width: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf(['small', 'medium', 'large'])
  ])
}
Container.defaultProps = {
  width: 'large'
}
