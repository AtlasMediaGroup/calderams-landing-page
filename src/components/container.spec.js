/**
 * Container component test
 * CalderaMS Landing page
 *
 * @license ATLASCommercial
 */

import React from 'react'
import { shallow } from 'enzyme'
import Container from './Container'

describe('Container', () => {
  it('renders correctly', () => {
    shallow(<Container />)
  })

  it('sets size to number to access ternary branch', () => {
    shallow(<Container width={5} />)
  })
})
