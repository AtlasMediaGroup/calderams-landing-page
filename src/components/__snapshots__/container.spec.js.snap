// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container component renders correctly 1`] = `
<div
  css={
    Array [
      Object {
        "@media (min-width: 412px)": Object {
          "paddingLeft": "2rem",
          "paddingRight": "2rem",
        },
        "boxSizing": "border-box",
        "marginLeft": "auto",
        "marginRight": "auto",
        "paddingLeft": "2rem",
        "paddingRight": "2rem",
        "position": "relative",
      },
      Object {
        "maxWidth": "69.6rem",
      },
    ]
  }
/>
`;
